Source/WebCore/platform/graphics/GraphicsContext.h

```diff
@@ -314,6 +316,9 @@ class GraphicsContext {
     virtual void drawLinesForText(const FloatPoint&, float thickness, const DashArray& widths, bool printing, bool doubleLines = false, StrokeStyle = StrokeStyle::SolidStroke) = 0;
     virtual void drawDotsForDocumentMarker(const FloatRect&, DocumentMarkerLineStyle) = 0;
 
+    // DisplayList
+    WEBCORE_EXPORT virtual void drawDisplayListItems(const Vector<DisplayList::Item>&, const DisplayList::ResourceHeap&, const FloatPoint& destination);
```

Comment:
It feels a bit strange to pass in a `ResourceHeap` to a `GraphicsContext` drawing function, like it's almost an implementation detail that the `GraphicsContext` layer shouldn't need to know about. But I don't have a better suggestion right now, since changing this to take a `DisplayList` instead would mean that when replaying in the GPU process, we'd need to construct a `DisplayList` from the recorded items plus a reference to the destination's `ResourceHeap`. But `DisplayList` only supports holding an owned copy of the resource heap.

Maybe in a future patch if you move towards display lists being a resource type themselves, we could have the reusable display list own and transfer its resource heap, instead of recording resource use in the target display list.